utils/diskusage/disk_usage.go

```diff
@@ -7,39 +7,32 @@ import "syscall"
 
 // DiskUsage contains usage data and provides user-friendly access methods.
 type DiskUsage struct {
-	stat *syscall.Statfs_t
+	// Free is the total free bytes on file system.
```

Comment:
I refactored this to:
1. return an error if the call failed (where previously it would swallow the error) and to 
2. return a struct in the success case which makes it easier to test how the data deletion business logic works for a given file system state without needing to create that file system state in reality (which can be difficult to do in tests).